File: 14.Longest_Common_Prefix/Longest_Common_Prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if(strs.size() <= 0){
            return "";
        }
        int min_len = INT_MAX;
        for(int i=0;i<strs.size();i++){
            min_len = min(int(strs[i].length()), min_len);
        }
        for(int i=0;i<min_len;i++){
            int val = strs[0][i];
            for(int j=1;j<strs.size();j++){
                if(strs[j][i] != val){
                    return strs[0].substr(0,i);
                }
            }
        }
        return strs[0].substr(0, min_len);
    }
};

File: 14.Longest_Common_Prefix/Longest_Common_Prefix.java
public class Solution {
    public String longestCommonPrefix(String[] strs) {
        if(strs.length <= 0){
            return "";
        }
        int min_len = Integer.MAX_VALUE;
        for(int i=0;i<strs.length;i++){
            min_len = Math.min(strs[i].length(), min_len);
        }
        for(int i=0;i<min_len;i++){
            char val = strs[0].charAt(i);
            for(int j=1;j<strs.length;j++){
                if(strs[j].charAt(i) != val){
                    return strs[0].substring(0, i);
                }
            }
        }
        return strs[0].substring(0,min_len);
    }
}